opm/parser/eclipse/Parser/MessageContainer.hpp

```diff
@@ -20,17 +20,27 @@
 #ifndef MESSAGECONTAINER_H
 #define MESSAGECONTAINER_H
 
-#endif // OPM_MESSAGECONTAINER_HEADER_INCLUDED
-
 #include <string>
 #include <vector>
 #include <memory>
 
 namespace Opm {
 
-    namespace MessageType {
+    struct Location {
+        Location() = default;
+        Location( const std::string& fn, int ln ) :
+            filename( fn ), lineno( ln ) {}
```

Comment:
On Tue, Apr 05, 2016 at 01:45:05PM -0700, Joakim Hove wrote:
> >  #include <string>
> >  #include <vector>
> >  #include <memory>
> >  
> >  namespace Opm {
> >  
> > -    namespace MessageType {
> > +    struct Location {
> > +        Location() = default;
> > +        Location( const std::string& fn, int ln ) :
> > +            filename( fn ), lineno( ln ) {}
> 
> This should throw for `ln <  0` - it should be freaking impossible to explicitly create a `Location` instance which evaluates to `false`.

or instead change the type from int to std::size_t or another
unsgigned integer as negative values make no sense at all